cdap-api/src/main/java/co/cask/cdap/api/schedule/ScheduleBuilder.java

```diff
@@ -154,33 +158,33 @@
    * @param program the name of the program
    * @param programStatuses the set of statuses to trigger the schedule. The schedule will be triggered if the status of
    *                        the specific program transitioned to one of these statuses.
-   * @return this {@link ScheduleBuilder}
+   * @return a {@link ScheduleCreationSpec}
    */
-  ScheduleCreationBuilder triggerOnProgramStatus(String programNamespace, String application, String appVersion,
-                                                 ProgramType programType, String program,
-                                                 ProgramStatus... programStatuses);
+  ScheduleCreationSpec triggerOnProgramStatus(String programNamespace, String application, String appVersion,
+                                              ProgramType programType, String program,
+                                              ProgramStatus... programStatuses);
 
   /**
    * Creates a schedule which is triggered in the same application version.
    *
-   * @see ScheduleBuilder#triggerOnProgramStatus(String, String, ProgramType, String, ProgramStatus...)
+   * @see #triggerOnProgramStatus(String, String, ProgramType, String, ProgramStatus...)
    */
-  ScheduleCreationBuilder triggerOnProgramStatus(String programNamespace, String application, ProgramType programType,
-                                                 String program, ProgramStatus... programStatuses);
+  ScheduleCreationSpec triggerOnProgramStatus(String programNamespace, String application, ProgramType programType,
+                                              String program, ProgramStatus... programStatuses);
 
   /**
    * Creates a schedule which is triggered in the same application and application version.
    *
-   * @see ScheduleBuilder#triggerOnProgramStatus(String, String, ProgramType, String, ProgramStatus...)
+   * @see #triggerOnProgramStatus(String, String, ProgramType, String, ProgramStatus...)
    */
-  ScheduleCreationBuilder triggerOnProgramStatus(String programNamespace, ProgramType programType, String program,
-                                                 ProgramStatus... programStatuses);
+  ScheduleCreationSpec triggerOnProgramStatus(String programNamespace, ProgramType programType, String program,
```

Comment:
Same comment as above. Not sure when users want to specify a program in an app in another namespace with same name and version